packages/node_modules/nav-frontend-skjema/src/valid-input.js

```diff
@@ -0,0 +1,286 @@
+    }
+
+    isValid() {
+        return this.state.valid;
+    }
+
+    validate() {
+        const result = this.runValidation();
+        this.props.onValidate(result);
+    }
+
+    runValidation() {
+        let valid = true;
+        const testsCopy = this.validators.map((validator) => {
+            const validatorResult = validator();
+            if (!validatorResult.verdict) valid = false;
+            return validatorResult;
+        });
+
+        this.setState({
+            tests: testsCopy.slice(),
+            valid
+        });
+
+        return {
+            name: this.props.name,
+            tests: testsCopy.slice(),
+            valid
+        };
+    }
+
+    render() {
+        const {
+            onValidate, onChange, onBlur, validateOnChange, validateOnBlur, type, // eslint-disable-line no-unused-vars
+            matchPattern, component, optional, feil, ...other // eslint-disable-line no-unused-vars
+        } = this.props;
+
+        const failedVerdict = (!this.state.valid) ? { feilmelding: this.getFirstFailedVerdict().type } : undefined;
+
+        if (this.supportedInputTextTypes.some((t) => t === type)) {
+            return (
+                <Input
+                    inputRef={(node) => { this.element = node; }}
+                    onChange={this.onChange}
+                    onBlur={this.onBlur}
+                    feil={feil || failedVerdict}
+                    {...other}
+                />
+            );
+        } else if (this.supportedInputNumberTypes.some((t) => t === type)) {
+            return (
+                <Input
+                    inputRef={(node) => { this.element = node; }}
+                    onChange={this.onChange}
+                    onBlur={this.onBlur}
+                    feil={feil || failedVerdict}
+                    type="number"
+                    {...other}
+                />
+            );
+        } else if (type === 'textarea') {
+            return (
+                <Textarea
+                    textareaRef={(node) => { this.element = node; }}
+                    onChange={this.onChange}
+                    onBlur={this.onBlur}
+                    feil={feil || failedVerdict}
+                    {...other}
+                />
+            );
+        } else if (type === 'select') {
+            return (
+                <Select
+                    selectRef={(node) => { this.element = node; }}
+                    onChange={this.onChange}
+                    onBlur={this.onBlur}
+                    feil={feil || failedVerdict}
+                    {...other}
+                >
+                    {this.props.children}
+                </Select>
+            );
+        }
+
+        return 'Unsupported input type "' + type + '"'; // eslint-disable-line prefer-template
+    }
+}
+
+ValidInput.propTypes = {
+    onValidate: PT.func,
+    onChange: PT.func,
+    onBlur: PT.func,
+    type: PT.oneOf(
```

Comment:
Generelt ville jeg nok ha begrenset type-attributtet til å bruke reelle html input types :) om vi spesifiserer accountnr, orgnr og lignende på ValidInput sine types og enforcer valideringsregler basert på det, tenker jeg det kan potensielt bli litt utydelig hva slags funksjon det egentlig har å sette på en gitt type på et inputfelt.

Reply:
👍 
Jeg tenker det er en god ide og holde APIet på denne komponenten så likt `input` sitt apiet som mulig, og heller lage egne wrappere for `textarea`, `select` etc.